Cards/CardBase.gd

```diff
@@ -46,6 +48,9 @@ func on_card_play(caster: Entity, targets: Array[Entity]) -> void:
 	#Split up targeted attacks and all attacks
 	var effects_targeting_single: Array[EffectData] = get_effects_targeting_single()
 	var effects_targeting_all: Array[EffectData] = get_effects_target_all()
+
+	#uses the energy asocieted with the card
+	PlayerManager.player.get_energy_component().use_energy(self)	
```

Comment:
Maybe add a `if caster is Player` before this call, since enemies use `on_card_play` too

Reply:
good catch